lib/Basics/files.cpp

```diff
@@ -2537,7 +2546,7 @@ int TRI_CreateDatafile(std::string const& filename, size_t maximalSize) {
   }
 
   // no fallocate present, or at least pretend it's not there...
-  int res = TRI_ERROR_NOT_IMPLEMENTED;
+  int res = 1;
```

Comment:
while this line looks ok, a few lines further down (after the #ifdef) we are comparing the numeric `res` variable against an ErrorCode (TRI_ERROR_NO_ERROR).
Shouldn't that line be `if (res != 0) {` instead?

Reply:
Right. I'll change it here, but there are most probably more places like that. The compiler will catch these as soon as I remove the implicit cast `operator int()` in `ErrorCode`.